examples/plot_linear_regression.py

```diff
@@ -0,0 +1,217 @@
+        then the shape of each of the arrays will be
+        (n_channels, n_timepoints).
+    """
+    if names is None:
+        names = ['x%i' % i for i in range(design_matrix.shape[1])]
+
+    if isinstance(inst, BaseEpochs):
+        picks = pick_types(inst.info, meg=True, eeg=True, ref_meg=True,
+                           stim=False, eog=False, ecg=False,
+                           emg=False, exclude=['bads'])
+        if [inst.ch_names[p] for p in picks] != inst.ch_names:
+            warn('Fitting linear model to non-data or bad channels. '
+                 'Check picking')
+        msg = 'Fitting linear model to epochs'
+        data = inst.get_data()
+        out = EvokedArray(np.zeros(data.shape[1:]), inst.info, inst.tmin)
+    elif isgenerator(inst):
+        msg = 'Fitting linear model to source estimates (generator input)'
+        out = next(inst)
+        data = np.array([out.data] + [i.data for i in inst])
+    elif isinstance(inst, list) and isinstance(inst[0], SourceEstimate):
+        msg = 'Fitting linear model to source estimates (list input)'
+        out = inst[0]
+        data = np.array([i.data for i in inst])
+    else:
+        raise ValueError('Input must be epochs or iterable of source '
+                         'estimates')
+    logger.info(msg + ', (%s targets, %s regressors)' %
+                (np.product(data.shape[1:]), len(names)))
+
+    # calls refurbished function _fit_lm with sklearn estimator
+    lm_params = _fit_lm_reloaded(data, design_matrix, names)
+    lm = namedtuple('lm', 'beta stderr t_val p_val mlog10_p_val')
+    lm_fits = {}
+
+    for name in names:
+        parameters = [p[name] for p in lm_params]
+        for ii, value in enumerate(parameters):
+            out_ = out.copy()
+            if not isinstance(out_, (SourceEstimate, Evoked)):
+                raise RuntimeError('Invalid container.')
+            out_._data[:] = value
+            parameters[ii] = out_
+        lm_fits[name] = lm(*parameters)
+    logger.info('Done')
+    return lm_fits
+
+
+# change mne _fit_lm function
+def _fit_lm_reloaded(data, design_matrix, names):
+    """Aux function."""
+    from scipy import stats
+    n_samples = len(data)
+    n_features = np.product(data.shape[1:])
+    if design_matrix.ndim != 2:
+        raise ValueError('Design matrix must be a 2d array')
+    n_rows, n_predictors = design_matrix.shape
+
+    if n_samples != n_rows:
+        raise ValueError('Number of rows in design matrix must be equal '
+                         'to number of observations')
+    if n_predictors != len(names):
+        raise ValueError('Number of regressor names must be equal to '
+                         'number of column in design matrix')
+
+    y = np.reshape(data, (n_samples, n_features))
+
+    # fit linear model with sklearn
+    # use fit intercept=false because b0 already in design matrix
+    linaer_model = LinearRegression(fit_intercept=False).fit(design_matrix, y)
```

Comment:
here I'm replycing `betas, resid_sum_squares, _, _ = linalg.lstsq(a=design_matrix, b=y)` with `linaer_model = LinearRegression(fit_intercept=False).fit(design_matrix, y)`

Reply:
oups, typo. I'll correct that in the next round